packages/website/cadl-prism-lang.js

```diff
@@ -0,0 +1,58 @@
+module.exports = {
+  comment: [
+    {
+      // multiline comments eg /* ASDF */
+      pattern: /(^|[^\\])\/\*[\s\S]*?(?:\*\/|$)/,
+      lookbehind: true,
+      greedy: true,
+    },
+    {
+      // singleline comments eg // ASDF
+      pattern: /(^|[^\\:])\/\/.*/,
+      lookbehind: true,
+      greedy: true,
+    },
+  ],
+
+  directives: { pattern: /#.*/g, greedy: true, alias: "comment" },
+
+  decorator: {
+    pattern: /@[$\w\xA0-\uFFFF]+/,
+    inside: {
+      at: {
+        pattern: /^@/,
+        alias: "operator",
+      },
+      function: /^[\s\S]+/,
+    },
+  },
+
+  property: {
+    pattern: /((?:^|[,{])[ \t]*)(?!\s)[_$a-zA-Z\xA0-\uFFFF](?:(?!\s)[$\w\xA0-\uFFFF])*(?=\s*:)/m,
+    lookbehind: true,
+    alias: "property",
+  },
+
+  string: [
+    {
+      pattern: /"""[^"][\s\S]*?"""/,
+      greedy: true,
+    },
+    {
+      pattern: /(^|[^\\"])"(?:\\.|\$(?!\{)|[^"\\\r\n$])*"/,
+      lookbehind: true,
+      greedy: true,
+    },
+  ],
+
+  boolean: /\b(?:false|true)\b/,
+  keyword:
+    /\b(?:import|model|namespace|op|interface|union|using|is|extends|enum|alias|return|void|never|if|else|projection)\b/,
+
+  function: /\b[a-z_]\w*(?=[ \t]*\()/i,
+
+  number: /(?:\b\d+(?:\.\d*)?|\B\.\d+)(?:E[+-]?\d+)?/i,
+  operator:
+    /--|\+\+|\*\*=?|=>|&&=?|\|\|=?|[!=]==|<<=?|>>>?=?|[-+*/%&|^!=<>]=?|\.{3}|\?\?=?|\?\.?|[~:]/,
+  punctuation: /[{}[\];(),.:]/,
+};
```

Comment:
I wonder if we can use the tmlanguage somehow? Seems annoying to have another set of regexes to maintain.

Reply:
Haha, yeah..., I did look if there was an existing way to auto convert it but doesn't seem like there is. Felt more work to get this to work.

Reply:
Maybe file an issue?

Reply:
oh you mean we are the one doing the syntax highlighting instead of using the extension

Reply:
Maybe? Just brainstorming here.

Reply:
Created an issue https://github.com/microsoft/cadl/issues/938

This was the PR that added bicep to PrismJS, where they seemed to have manually intepreted the tmlanguage and wrong a simple primsjs language
https://github.com/PrismJS/prism/pull/3027